src/test/java/com/faforever/client/map/generator/MapGeneratorServiceTest.java

```diff
@@ -0,0 +1,148 @@
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+  private MapGeneratorService instance;
+  @Mock
+  private PreferencesService preferencesService;
+  @Mock
+  private Preferences preferences;
+  @Mock
+  private ForgedAlliancePrefs forgedAlliancePrefs;
+  @Mock
+  private TaskService taskService;
+  @Mock
+  private ApplicationContext applicationContext;
+  @Mock
+  private DownloadMapGeneratorTask downloadMapGeneratorTask;
+  @Mock
+  private GenerateMapTask generateMapTask;
+
+  @Before
+  public void setUp() {
+    try {
+      customMapsDirectory.newFolder(testMapNameGenerator);//will be deleted on startup
+
+      fafDataDirectory.newFolder(MapGeneratorService.GENERATOR_EXECUTABLE_SUB_DIRECTORY);
+      Files.createFile(fafDataDirectory.getRoot().toPath().resolve(MapGeneratorService.GENERATOR_EXECUTABLE_SUB_DIRECTORY).resolve(String.format(MapGeneratorService.getGENERATOR_EXECUTABLE_FILENAME(), versionGeneratorPresent)));
+    } catch (IOException e) {
+      e.printStackTrace();
+      throw new RuntimeException("Could not create test map generator folder!");
+    }
+
+    when(preferencesService.getFafDataDirectory()).thenReturn(fafDataDirectory.getRoot().toPath());
+
+    when(preferencesService.getPreferences()).thenReturn(preferences);
+    when(preferences.getForgedAlliance()).thenReturn(forgedAlliancePrefs);
+    when(forgedAlliancePrefs.getCustomMapsDirectory()).thenReturn(customMapsDirectory.getRoot().toPath());
+
+
+    instance = new MapGeneratorService(applicationContext, preferencesService, taskService);
+
+    assertThat(Arrays.asList(Objects.requireNonNull(customMapsDirectory.getRoot().list())), not(contains(testMapNameGenerator)));
+
+    when(downloadMapGeneratorTask.getFuture()).thenReturn(CompletableFuture.completedFuture(null));
+    when(generateMapTask.getFuture()).thenReturn(CompletableFuture.completedFuture(null));
+    doAnswer(invocation -> {
+      @SuppressWarnings("unchecked")
+      CompletableTask<Void> task = invocation.getArgument(0);
+      WaitForAsyncUtils.asyncFx(task);
+      task.getFuture().get();
+      return task;
+    }).when(taskService).submitTask(any());
+  }
+
+
+  @Test
+  public void testGenerateMapNoGeneratorPresent() {
+    when(applicationContext.getBean(DownloadMapGeneratorTask.class)).thenReturn(downloadMapGeneratorTask);
+    when(applicationContext.getBean(GenerateMapTask.class)).thenReturn(generateMapTask);
+
+    instance.generateMap(testMapNameNoGenerator).join();
+
+    verify(taskService).submitTask(downloadMapGeneratorTask);
+    verify(downloadMapGeneratorTask).setVersion(versionNoGeneratorPresent);
+
+    //See test below
+    verify(taskService).submitTask(generateMapTask);
+  }
+
+  @Test
+  public void testGenerateMapGeneratorPresent() {
+    when(applicationContext.getBean(GenerateMapTask.class)).thenReturn(generateMapTask);
+
+    instance.generateMap(testMapNameGenerator).join();
```

Comment:
It is a CompletableFuture<String> please use get to also verify the result or if it has none then make it Void